### PR TITLE
Support Ruby 3.0 & Rails 6.1

### LINKED ISSUE #585: Wrong number of arguments (given 4, expected 3) when running migrations
**Describe the bug**
When running the ActiveRecord migrations in a new rails app, the migrations fail with error `wrong number of arguments (given 4, expected 3)`.

**To Reproduce**
1. Create a new rails app (`rails new rpush-example`)
2. Add rpush to the gemfile (`gem 'rpush'`)
3. Run `bundle install`
4. Run `bundle exec rpush init`
5. Choose to install the ActiveRecord migrations (`y`)
6. Run `bundle exec rake db:migrate`
7. See error `wrong number of arguments (given 4, expected 3)`

**Expected behavior**
The migrations succeed without errors

**Logs and other output**
````
== 20210201124947 AddRpush: migrating =========================================
-- create_table(:rapns_notifications)
   -> 0.0014s
-- add_index(:rapns_notifications, [:delivered, :failed, :deliver_after], {:name=>"index_rapns_notifications_multi"})
   -> 0.0006s
-- create_table(:rapns_feedback)
   -> 0.0005s
-- add_index(:rapns_feedback, :device_token)
   -> 0.0007s
-- add_column(:rapns_notifications, :alert_is_json, :boolean, {:null=>true, :default=>false})
   -> 0.0007s
-- add_column(:rapns_notifications, :app, :string, {:null=>true})
   -> 0.0009s
-- add_column(:rapns_feedback, :app, :string, {:null=>true})
   -> 0.0005s
-- create_table(:rapns_apps)
   -> 0.0006s
-- add_column(:rapns_notifications, :type, :string, {:null=>true})
   -> 0.0005s
-- add_column(:rapns_apps, :type, :string, {:null=>true})
   -> 0.0005s
-- change_column(:rapns_notifications, :type, :string, {:null=>false})
   -> 0.0108s
-- change_column(:rapns_apps, :type, :string, {:null=>false})
   -> 0.0067s
-- change_column(:rapns_notifications, :device_token, :string, {:null=>true, :limit=>64})
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

wrong number of arguments (given 4, expected 3)
/home/andreas/apps/rpush-example/db/migrate/20210201124947_add_rpush.rb:167:in `up'
/home/andreas/apps/rpush-example/db/migrate/20210201124947_add_rpush.rb:31:in `map'
/home/andreas/apps/rpush-example/db/migrate/20210201124947_add_rpush.rb:31:in `up'
/home/andreas/.asdf/installs/ruby/3.0.0/bin/bundle:23:in `load'
/home/andreas/.asdf/installs/ruby/3.0.0/bin/bundle:23:in `<main>'

Caused by:
ArgumentError: wrong number of arguments (given 4, expected 3)
/home/andreas/apps/rpush-example/db/migrate/20210201124947_add_rpush.rb:167:in `up'
/home/andreas/apps/rpush-example/db/migrate/20210201124947_add_rpush.rb:31:in `map'
/home/andreas/apps/rpush-example/db/migrate/20210201124947_add_rpush.rb:31:in `up'
/home/andreas/.asdf/installs/ruby/3.0.0/bin/bundle:23:in `load'
/home/andreas/.asdf/installs/ruby/3.0.0/bin/bundle:23:in `<main>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
````

**System configuration (please complete the following information):**
 - OS: Ubuntu
 - OS version: 20.04.1 LTS
 - Ruby version: ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-linux]
 - Rails version: 6.1.1
 - Rpush version: 5.3.0

**Additional context**
This line runs fine:
```ruby
change_column :rapns_apps, :type, :string, null: false
```
But this one fails:
```ruby
change_column :rapns_notifications, :device_token, :string, { null: true, limit: 64 }
```
Changing it like this works:
```ruby
change_column :rapns_notifications, :device_token, :string, null: true, limit: 64
```

Could it be because of the separation of positional and keyword arguments in Ruby 3.0?

### PR DESCRIPTION
**Pull request by andreaslillebo**
Fixes errors on Ruby 3.0 & Rails 6.1, and tests these versions on CI.

Fixes #585

### CODE REVIEW

**Comment by andreaslillebo** [comment]
This should now be working.

@benlangfeld, could you please review this and let me know if any changes are needed?

**Review by benlangfeld — APPROVED** [review]

**Comment by benlangfeld** [comment]
I'd like to see what @aried3r thinks but this looks good to me.

**Comment by aried3r** [comment]
LGTM, I only have one question, why were the changes to the spec necessary?

File: spec/unit/daemon/pushy/delivery_spec.rb [hunk 1]
```diff
@@ -61,10 +61,12 @@
       it_behaves_like 'process notification'
     end
 
-    shared_examples 'retry delivery' do |response_code:|
-      let(:response_code) { response_code }
+    shared_examples 'retry delivery' do |options|
```
Comment (andreaslillebo) [inline spec/unit/daemon/pushy/delivery_spec.rb:64]: In Ruby 3.0, this way of destructuring arguments is no longer supported.

See the following link for more details:
https://bugs.ruby-lang.org/issues/14183#note-114

Comment (aried3r) [reply spec/unit/daemon/pushy/delivery_spec.rb:64]: Oh, that's an interesting discussion. I haven't seen it before. Thanks for the link!

**Pull request merged**